### PR TITLE
Feature/#38

### PR DESCRIPTION
**Pull request by xUnholy**
Fixes: #38

### CODE REVIEW

File: test/docker-compose.yml [hunk 1]
```diff
@@ -41,6 +47,6 @@ services:
       timeout: 5s
       retries: 5
 
-x-kube-compose:
-  push_images:
-    docker_registry: 'localhost:5000'
\ No newline at end of file
+# x-kube-compose:
```
Comment (xUnholy) [inline test/docker-compose.yml]: please uncomment

File: pkg/config/schema.go [hunk 2]
```diff
@@ -191,6 +191,33 @@ func (t *environment) Decode(into mapdecode.Into) error {
...
+	File    *string `mapdecode:"file"`
+	Service string  `mapdecode:"service"`
+}
+
+type extends struct {
+	File    *string
+	Service string
+}
+
+func (e *extends) Decode(into mapdecode.Into) error {
```
Comment (xUnholy) [inline pkg/config/schema.go]: unused function

Comment (jbrekelmans) [reply pkg/config/schema.go]: Used by mapdecode package

File: pkg/config/merge.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// extendedService is merged into service
+func merge(service *Service, extendedService *Service) {
+	// rules based on https://docs.docker.com/compose/extends/#adding-and-overriding-configuration
+	mergeMap(service.Environment, extendedService.Environment)
+	if len(extendedService.Image) > 0 {
+		service.Image = extendedService.Image
+	}
+}
+
+func mergeMap(env map[string]string, extendedEnv map[string]string) {
```
Comment (jbrekelmans) [inline pkg/config/merge.go]: Rename to mergeStringMap and rename args

File: pkg/config/merge.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	for k, v := range extendedEnv {
+		env[k] = v
```
Comment (jbrekelmans) [inline pkg/config/merge.go]: Only set `env[k]` if k is not present in env

File: pkg/config/merge.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package config
+
+// extendedService is merged into service
+func merge(service *Service, extendedService *Service) {
+	// rules based on https://docs.docker.com/compose/extends/#adding-and-overriding-configuration
+	mergeMap(service.Environment, extendedService.Environment)
+	if len(extendedService.Image) > 0 {
+		service.Image = extendedService.Image
```
Comment (jbrekelmans) [inline pkg/config/merge.go]: Image is only extended in docker-compose V1, apparently: https://github.com/docker/compose/blob/master/compose/config/config.py#L1128

Comment (russ-c) [reply pkg/config/merge.go]: will remove this for now

File: pkg/config/merge.go (hunk 4 above)
Comment (russ-c) [reply pkg/config/merge.go]: done

File: pkg/config/schema.go (hunk 2 above)
Comment (russ-c) [reply pkg/config/schema.go]: commented

File: test/docker-compose.yml (hunk 1 above)
Comment (russ-c) [reply test/docker-compose.yml]: done

File: pkg/config/merge.go (hunk 4 above)
Comment (russ-c) [reply pkg/config/merge.go]: done

File: pkg/config/merge.go (hunk 3 above)
Comment (russ-c) [reply pkg/config/merge.go]: done

File: pkg/config/merge.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+// extendedService is merged into service
+func merge(service *Service, extendedService *Service) {
+	// rules based on https://docs.docker.com/compose/extends/#adding-and-overriding-configuration
+	mergeStringMap(service.Environment, extendedService.Environment)
+	// TODO https://github.com/jbrekelmans/kube-compose/issues/48
+}
+
+func mergeStringMap(intoStringMap map[string]string, fromStringMap map[string]string) {
+	for k, v := range fromStringMap {
+		if _, ok := intoStringMap[k]; ok {
```
Comment (jbrekelmans) [inline pkg/config/merge.go]: Invert if: `!ok`

**Pull request merged**